### PR TITLE
Initial Rebrand Effort (rebased to master)

### PR DESCRIPTION
**Pull request by mauricioszabo**
Rebase of PR https://github.com/pulsar-edit/pulsar/pull/21 - original message below:

Large PR - Individual commits per associated segment

Rebrand all of the non-api/non-variable/non-github link instances of Atom or atom to Pulsar or pulsar respectively

Current build commands

nvm install 16.16.0 #May not be needed
npm install --location=global yarn
yarn install
yarn build
yarn build:apm #Not strictly necessary
yarn start
Known bugs as of initial PR:

GitHub renderer.html/renderer.js require is not defined
Tree-sitter is an incompatible version. It wants 13, gets 10
Settings-View package-manager fails to fetch local or outdated packages/themes - This is presumed to be a problematic interaction with apm. Apm expects files to be in a .atom directory, which is no longer the case.

### CODE REVIEW

File: src/main-process/atom-application.js [hunk 1]
```diff
@@ -1077,7 +1077,7 @@ module.exports = class AtomApplication extends EventEmitter {
 
   initializeAtomHome(configDirPath) {
     if (!fs.existsSync(configDirPath)) {
-      const templateConfigDirPath = fs.resolve(this.resourcePath, 'dot-atom');
+      const templateConfigDirPath = fs.resolve(this.resourcePath, 'templates');
```
Comment (Spiker985) [inline src/main-process/atom-application.js:1080]:
```suggestion
      const templateConfigDirPath = fs.resolve(this.resourcePath, 'dot-atom');
```
I had renamed dot-atom to templates to make it more concise as to what it was supposed to be, however, that doesn't belong in this PR.

We can revert this line

File: templates/.gitignore [hunk 2]
```diff
@@ -5,4 +5,4 @@ storage
 .apm
 .node-gyp
 .npm
-.atom-socket-secret-*
+.pulsar-socket-secret-*
```
Comment (Spiker985) [inline templates/.gitignore:8]: We'll need to revert the folder rename in this commit, but keep the file content changes

File: script/lib/copy-assets.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ module.exports = function() {
   console.log(`Copying assets to ${CONFIG.intermediateAppPath}`);
   let srcPaths = [
     path.join(CONFIG.repositoryRootPath, 'benchmarks', 'benchmark-runner.js'),
-    path.join(CONFIG.repositoryRootPath, 'dot-atom'),
+    path.join(CONFIG.repositoryRootPath, 'templates'),
```
Comment (Spiker985) [inline script/lib/copy-assets.js:16]:
```suggestion
    path.join(CONFIG.repositoryRootPath, 'dot-atom'),
```
Revert this line to coincide with reverting templates -> dot-atom

File: script/lib/lint-coffee-script-paths.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ const CONFIG = require('../config');
 
 module.exports = function() {
   const globPathsToLint = [
-    path.join(CONFIG.repositoryRootPath, 'dot-atom/**/*.coffee'),
+    path.join(CONFIG.repositoryRootPath, 'templates/**/*.coffee'),
```
Comment (Spiker985) [inline script/lib/lint-coffee-script-paths.js:12]:
```suggestion
    path.join(CONFIG.repositoryRootPath, 'dot-atom/**/*.coffee'),
```
Revert this line to coincide with reverting templates -> dot-atom

File: script/electron-builder.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
+const path = require('path')
```
Comment (Spiker985) [inline script/electron-builder.js:1]: This was accidentally pulled in from another branch during an errant rebase and should not be part of this PR. Apologies

File: src/protocol-handler-installer.js [hunk 6]
```diff
@@ -87,7 +87,7 @@ module.exports = class ProtocolHandlerInstaller {
         dismissable: true,
         icon: 'link',
         description:
-          'Atom is not currently set as the default handler for atom:// URIs. Would you like Atom to handle ' +
+          'Pulsar is not currently set as the default handler for atom:// URIs. Would you like Atom to handle ' +
```
Comment (Spiker985) [inline src/protocol-handler-installer.js:90]:
```suggestion
          'Pulsar is not currently set as the default handler for atom:// URIs. Would you like Pulsar to handle ' +
```

**Review by Spiker985 — CHANGES_REQUESTED** [review]
Requested updates to review `dot-atom` -> `templates`

File: script/electron-builder.js (hunk 5 above)
Comment (Spiker985) [reply script/electron-builder.js:1]: We decided that this script can stay as of now, as it won't make a difference at this moment, is up to date, and is expected to be required later

File: Dockerfile [hunk 7]
```diff
@@ -21,8 +21,8 @@ RUN apt-get update && \
         npm && \
     rm -rf /var/lib/apt/lists/*
 
-# Update npm and dependencies
-RUN npm install -g npm --loglevel error
+COPY . /pulsar
```
Comment (softcode589) [inline Dockerfile:24]: Why is the dockerfile being changed? This is a full rebrand effort. I know about the copy and workdir being changed to pulsar, but we could have some npm packages being maintained.

**Review by confused-Techie — APPROVED** [review]
This PR has been sitting open for quite some time. But many of the commits on this one I helped author, (or slightly edit really). But otherwise this looks good to me and I'll approve. 

This finally gets all the fantastic rebrand efforts over to master, so hopefully we could have a rebranding build from cirrus? @mauricioszabo 

But otherwise lets try to get a few more eyes on this, as well as some issues open on this PR @pulsar-edit/core

**Review by kaosine — APPROVED** [review]
This seems to be mostly right, and I'd recommend merging it, so we can merge the e12 one as well. Though do wish we could make the pulsar mentions more generic in the code so that anyone could fork and use this in their own projects. But I'd be happy to just get this done and gone, so we can move forward.

**Pull request merged**